src/components/UpdatePassword/UpdatePasswordButton.tsx

```diff
@@ -0,0 +1,127 @@
+import { useContext, useRef, useState } from "react";
+import { updatepasswordService } from "../../services/auth/updatepassword.service";
+import { FieldValues, useForm } from "react-hook-form";
+import { AuthContext } from "../../context/AuthContext";
+import toast from "react-hot-toast";
+import { Dialog } from "../../components/Dialog";
+import { ButtonSubmit } from "../Button/ButtonSubmit";
+
+export function UpdatePassword() {
+  const { session } = useContext(AuthContext);
+  const [showModal, setShowModal] = useState(false);
+  const formRef = useRef<HTMLFormElement>(null);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm();
+
+  interface UpdatepasswordRequest {
+    oldPassword: string;
+    newPassword: string;
+    token: string;
+  }
+
+  const closeModal = () => setShowModal(false);
+
+  const onSubmit = async (formData: FieldValues, token: string) => {
+    const req: UpdatepasswordRequest = {
+      oldPassword: formData.password,
+      newPassword: formData.newPassword,
+      token: token
+    };
+    if (!errors.password && !errors.newPassword) {
+      const response = await updatepasswordService(req);
+      if (response.success === true) {
+        toast.success("Password updated successfully");
+        setShowModal(false);
+      } else {
+        toast.error("Failed to update password");
+      }
+      formRef.current?.reset();
```

Comment:
It seems that this is not working as expected: 


https://github.com/hawks-atlanta/frontend-react/assets/62714297/d49bf9f4-9778-4e4e-9036-aae1a832d8a8